src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -991,6 +993,146 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase {
+
+      val q = spark.readStream
+        .format("delta")
+        .option("startingVersion", "0")
+        .load(inputDir.getCanonicalPath)
+        .writeStream
+        .format("delta")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start(outputDir.getCanonicalPath)
+      try {
+        q.processAllAvailable()
+        checkAnswer(
+          spark.read.format("delta").load(outputDir.getAbsolutePath),
+          (0 until 20).map(_.toLong).toDF())
+      } finally {
+        q.stop()
+      }
+
+      generateCommits(tblLoc, start + 40.minutes)
+
+      val q2 = spark.readStream
+        .format("delta")
+        .option("startingTimestamp", "2020-07-10 15:20:40")
+        .load(inputDir.getCanonicalPath)
+        .writeStream
+        .format("delta")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start(outputDir.getCanonicalPath)
+      try {
+        q2.processAllAvailable()
+        checkAnswer(
+          spark.read.format("delta").load(outputDir.getAbsolutePath),
+          (0 until 30).map(_.toLong).toDF())
+      } finally {
+        q2.stop()
+      }
+    }
+  }
+
+  test("startingVersion: invalid inputs") {
+    withTempDirs { (inputDir, outputDir, checkpointDir) =>
+      val tblLoc = inputDir.getCanonicalPath
+      val start = 1594418440842L
+      generateCommits(tblLoc, start, start + 20.minutes)
+
+      val e = intercept[StreamingQueryException] {
+        spark.readStream
+          .format("delta")
+          .option("startingVersion", "-1")
+          .load(inputDir.getCanonicalPath)
+          .writeStream
+          .format("delta")
+          .option("checkpointLocation", checkpointDir.getCanonicalPath)
+          .start(outputDir.getCanonicalPath)
+          .processAllAvailable()
+      }
+
+      assert(e.getCause.isInstanceOf[IllegalArgumentException])
+      assert(e.getCause.getMessage.contains("Invalid value '-1' for option 'startingVersion'"))
+
+      val e2 = intercept[StreamingQueryException] {
+        spark.readStream
+          .format("delta")
+          .option("startingVersion", "5")
+          .load(inputDir.getCanonicalPath)
+          .writeStream
+          .format("delta")
+          .option("checkpointLocation", checkpointDir.getCanonicalPath)
+          .start(outputDir.getCanonicalPath)
+          .processAllAvailable()
+      }
+
+      assert(e2.getCause.isInstanceOf[AnalysisException])
+      assert(e2.getCause.getMessage.contains("Cannot time travel Delta table to version 5"))
+    }
+  }
+
+  test("startingTimestamp: invalid inputs") {
+    withTempDirs { (inputDir, outputDir, checkpointDir) =>
+      val tblLoc = inputDir.getCanonicalPath
+      val start = 1594418440842L
+      generateCommits(tblLoc, start, start + 20.minutes)
+
+      val e = intercept[StreamingQueryException] {
+        spark.readStream
+          .format("delta")
+          .option("startingTimestamp", "2019-07-10 15:20:40")
```

Comment:
what error do we throw if we give a bad timestamp? Also can we also please test `"2019-07-10"`